ssh-server/main.go

```diff
@@ -113,38 +118,102 @@ func serviceChannel(newChannel ssh.NewChannel) error {
 	}
 	defer channel.Close()
 
-	// Start the command with a pseudo-terminal.
-	shell, err := pty.Start(exec.Command("bash"))
+	allow := map[string]bool{"shell": true, "exec": true, "pty-req": true, "window-change": true}
+	resize := make(chan *pty.Winsize, 1)
+	defer close(resize)
+	done := make(chan struct{})
+	for {
+		select {
+		case <-done:
+			return nil
+		case req, ok := <-requests:
+			if !ok {
+				return nil
+			}
+			if req.WantReply {
+				req.Reply(allow[req.Type], nil)
+			}
+			switch req.Type {
+			case "pty-req":
+				payload := req.Payload
```

Comment:
Use bytes.NewReader right here and pass that around instead of the slice.  That will clean this up a lot.

Reply:
done